openmc/model/model.py

```diff
@@ -1259,11 +1265,20 @@ def differentiate_mats(self, diff_volume_method: str = None, depletable_only: bo
         for cell in self.geometry.get_all_material_cells().values():
             if cell.fill in distribmats:
                 mat = cell.fill
-                if diff_volume_method != 'match cell':
+
+                # Clone materials
+                if cell.num_instances > 1:
                     cell.fill = [mat.clone() for _ in range(cell.num_instances)]
-                elif diff_volume_method == 'match cell':
+                else:
                     cell.fill = mat.clone()
-                    cell.fill.volume = cell.volume
+
+                # For 'match cell', assign volumes based on the cells
+                if diff_volume_method == 'match cell':
+                    if cell.fill_type == 'distribmat':
+                        for clone_mat in cell.fill:
+                            clone_mat.volume = cell.volume
```

Comment:
Kinda late to the party here, but should the volume assigned to the material be divided by the number of cell instances in this line? Am I missing something?